DirectProgramming/DPC++FPGA/ReferenceDesigns/db/src/query11/pipe_types.hpp

```diff
@@ -0,0 +1,129 @@
+#ifndef __PIPE_TYPES_H__
+#define __PIPE_TYPES_H__
+#pragma once
+
+#include <CL/sycl.hpp>
+#include <CL/sycl/INTEL/fpga_extensions.hpp>
+
+#include "../db_utils/StreamingData.hpp"
+#include "../dbdata.hpp"
+
+using namespace sycl;
+
+//
+// A single row of the SUPPLIER table
+// with a subset of the columns (needed for this query)
+//
+class SupplierRow {
+ public:
+  // SupplierRow() : valid(false), suppkey(0), nationkey(0) {}
+  SupplierRow() {}
+  SupplierRow(bool _valid, DBIdentifier _suppkey, unsigned char _nationkey)
```

Comment:
Leading underscore not preferred here.   Since these are public members, OK not to have trailing underscore.   Maybe the right approach is v_valid, v_suppkey, ... (v_ == value?)